### PR TITLE
Shared preference for notification added

### PR DESCRIPTION
**Pull request by gbielanski**
Please review

### CODE REVIEW

File: app/src/main/java/pl/example/android/garbageapp/ui/ActivitySectorYellow.java [hunk 1]
```diff
@@ -37,10 +38,13 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            binding.notificationSwitch.setChecked(true);
+
         //TODO mViewModel observe and bind data to UI
     }
 
-    public void showToast(View view) {
-        Toast.makeText(this, "Przypomnienie ustawione dla STREFY III", Toast.LENGTH_LONG).show();
+    public void setNotification(View view) {
+        super.setNotification(view);
     }
```
Comment (miltomasz) [inline app/src/main/java/pl/example/android/garbageapp/ui/ActivitySectorYellow.java]: When you declare setNotification() method as public method in BaseActivitySector, won't it work?

File: app/src/main/java/pl/example/android/garbageapp/ui/ActivitySectorGreen.java [hunk 2]
```diff
@@ -37,10 +38,13 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            binding.notificationSwitch.setChecked(true);
+
         //TODO mViewModel observe and bind data to UI
     }
 
-    public void showToast(View view) {
-        Toast.makeText(this, "Przypomnienie ustawione dla STREFY I", Toast.LENGTH_LONG).show();
+    public void setNotification(View view) {
+        super.setNotification(view);
     }
```
Comment (miltomasz) [inline app/src/main/java/pl/example/android/garbageapp/ui/ActivitySectorGreen.java]: When you declare setNotification() method as public method in BaseActivitySector, won't it work?

File: app/src/main/java/pl/example/android/garbageapp/ui/BaseActivitySector.java [hunk 3]
```diff
@@ -25,4 +35,25 @@ protected void onCreate(Bundle savedInstanceState) {
         DetailViewModelFactory factory = InjectorUtils.provideDetailViewModelFactory(currentSector().getApplicationContext(), sectorType());
         mViewModel = ViewModelProviders.of(currentSector(), factory).get(DetailActivityViewModel.class);
     }
+
+    protected boolean checkIfSectorMarkedAsNotification(){
```
Comment (miltomasz) [inline app/src/main/java/pl/example/android/garbageapp/ui/BaseActivitySector.java]: I would change the name of the method to isMarkedForNotification().

File: app/src/main/java/pl/example/android/garbageapp/ui/BaseActivitySector.java [hunk 4]
```diff
@@ -25,4 +35,25 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        int notificationSectorType = SectorType.toInt(SectorType.UNSET);
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(currentSector());
+        if(sharedPreferences.contains(NOTIFICATION_SECTOR_TYPE))
+            notificationSectorType = sharedPreferences.getInt(NOTIFICATION_SECTOR_TYPE, notificationSectorType);
+
+        return notificationSectorType == SectorType.toInt(sectorType());
+    }
+
+    protected void setNotification(View view){
```
Comment (miltomasz) [inline app/src/main/java/pl/example/android/garbageapp/ui/BaseActivitySector.java]: The same as above - method name markForNotification() for example is more readable for me.

**Review by miltomasz — COMMENTED** [review]
I added a few comments mostly regarding names and naming.

File: app/src/main/java/pl/example/android/garbageapp/ui/ActivitySectorGreen.java (hunk 2 above)
Comment (gbielanski) [reply app/src/main/java/pl/example/android/garbageapp/ui/ActivitySectorGreen.java]: Good point. It was my bad day for finding names :)

File: app/src/main/java/pl/example/android/garbageapp/ui/BaseActivitySector.java (hunk 3 above)
Comment (gbielanski) [reply app/src/main/java/pl/example/android/garbageapp/ui/BaseActivitySector.java]: Good point. It was my bad day for finding names :)

File: app/src/main/java/pl/example/android/garbageapp/ui/BaseActivitySector.java (hunk 4 above)
Comment (gbielanski) [reply app/src/main/java/pl/example/android/garbageapp/ui/BaseActivitySector.java]: Good point. It was my bad day for finding names :)

File: app/src/main/java/pl/example/android/garbageapp/ui/ActivitySectorGreen.java (hunk 2 above)
Comment (gbielanski) [reply app/src/main/java/pl/example/android/garbageapp/ui/ActivitySectorGreen.java]: It was worked but warning was displayed. I found another approach.

File: app/src/main/java/pl/example/android/garbageapp/ui/ActivitySectorYellow.java (hunk 1 above)
Comment (gbielanski) [reply app/src/main/java/pl/example/android/garbageapp/ui/ActivitySectorYellow.java]: It was worked but warning was displayed. I found another approach.

**Review by miltomasz — APPROVED** [review]

**Pull request merged**